eventuals/task.h

```diff
@@ -389,12 +460,26 @@ struct _TaskFromToWith {
 // started via 'TaskFromToWith::Start()'. If used as a continuation
 // then it can't be moved after starting, just like all other
 // continuations.
-template <typename From_, typename To_, typename... Args_>
+template <
+    typename From_,
+    typename To_,
+    typename... Args_>
 class TaskFromToWith {
  public:
   template <typename Arg>
   using ValueFrom = To_;
 
+  TaskFromToWith(
+      std::conditional_t<
+          std::is_void_v<To_>,
+          std::monostate,
+          std::conditional_t<
+              !std::is_reference_v<To_>,
+              To_,
+              std::reference_wrapper<
+                  std::remove_reference_t<To_>>>> value)
```

Comment:
Seeing this one enough (and it's big enough) so let's give it a top-level name:

```cpp
template <typename T>
using MonostateIfVoidOrReferenceWrapperOr = std::conditional_t< ...>;
```

Then:

```suggestion
  TaskFromToWith(MonstateIfVoidOrReferenceWrapperOr<To_> value)
```